purchase_receipt_progress/models/purchase_order.py

```diff
@@ -0,0 +1,89 @@
+    qty_received = fields.Float()
+    to_receive = fields.Float()
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    purchased_quantity_total = fields.Float(
+        string="Total Quantity", compute="_compute_purchased_quantity_total"
+    )
+    delivery_status = fields.Float(compute="_compute_delivery_status")
+    received_by_category_ids = fields.One2many(
+        "purchase.order.received.by.category",
+        "order_id",
+        string="Received by Category",
+        compute="_compute_received_by_category_ids",
+    )
+
+    @api.depends("order_line.product_qty")
+    def _compute_purchased_quantity_total(self):
+        for order in self:
+            order.purchased_quantity_total = sum(order.mapped("order_line.product_qty"))
+
+    @api.depends("order_line.product_qty", "order_line.qty_received")
+    def _compute_delivery_status(self):
+        for order in self:
+            received_quantity = sum(order.mapped("order_line.qty_received"))
+            if order.purchased_quantity_total and received_quantity:
+                order.delivery_status = (
+                    received_quantity / order.purchased_quantity_total
+                )
+            else:
+                order.delivery_status = 0
+
+    @api.depends("order_line.product_qty", "order_line.qty_received")
+    def _compute_received_by_category_ids(self):
+        for order in self:
+            purchase_order_received_by_category_obj = self.env[
+                "purchase.order.received.by.category"
+            ]
+            order_id = (
+                order._origin.id if isinstance(order.id, models.NewId) else order.id
+            )
+            purchase_order_received_by_category_obj.search(
+                [("order_id", "=", order_id)]
+            ).unlink()
+            order.received_by_category_ids = False
+            received_by_category_ids = []
+            for category in order.mapped("order_line.product_id.categ_id"):
+                lines = order.order_line.filtered(
+                    lambda l: l.product_id.categ_id == category
+                )
+                if lines:
+                    values = {
+                        "order_id": order_id,
+                        "category_id": category.id,
+                        "product_qty": sum(lines.mapped("product_qty")),
+                        "qty_received": sum(lines.mapped("qty_received")),
+                        "to_receive": sum(lines.mapped("to_receive")),
+                    }
+                    received_by_category_id = (
+                        purchase_order_received_by_category_obj.create(values)
+                    )
+                    received_by_category_ids.append((4, received_by_category_id.id))
+            order.received_by_category_ids = received_by_category_ids
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    to_receive = fields.Float(compute="_compute_to_receive")
```

Comment:
```suggestion
    qty_to_receive = fields.Float(
        compute="_compute_to_receive",
        digits='Product Unit of Measure',
        store=True,
    )
```

* Naming: `qty_to_receive` is more consistent with existing field `qty_received`.
* Stored: as `qty_received` is stored, let's benefit this for reporting
* Digits: should be set for uom float fields